src/dist/replication/lib/replica_stub.cpp

```diff
@@ -1231,7 +1257,36 @@ void replica_stub::close()
         _log->close();
         _log = nullptr;
     }
+}
 
+std::string replica_stub::get_replica_dir(const char* app_type, global_partition_id gpid) const
```

Comment:
What's the meaning of "moving the replica directories on replica-stub initialization if changes are made to the data directories"? Could you explain it in details?